src/main/java/org/isf/patient/service/PatientIoOperationRepositoryImpl.java

```diff
@@ -90,15 +91,48 @@
 
 	private Predicate wordExistsInOneOfPatientFields(String word, CriteriaBuilder cb, Root<Patient> root) {
 		return cb.or(
-			cb.like(cb.lower(root.get("code").as(String.class)), like(word)),
-			cb.like(cb.lower(root.get("secondName").as(String.class)), like(word)),
-			cb.like(cb.lower(root.get("firstName").as(String.class)), like(word)),
-			cb.like(cb.lower(root.get("note").as(String.class)), like(word)),
-			cb.like(cb.lower(root.get("taxCode").as(String.class)), like(word))
+				cb.like(cb.lower(root.get("code").as(String.class)), like(word)),
+				cb.like(cb.lower(root.get("secondName").as(String.class)), like(word)),
+				cb.like(cb.lower(root.get("firstName").as(String.class)), like(word)),
+				cb.like(cb.lower(root.get("note").as(String.class)), like(word)),
+				cb.like(cb.lower(root.get("taxCode").as(String.class)), like(word))
 		);
 	}
 
 	private String like(String word) {
 		return "%" + word + "%";
 	}
+
+	public List<Patient> getPatientsByParams(Map<String, Object> params) {
+
+		CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+		CriteriaQuery<Patient> query = cb.createQuery(Patient.class);
+		Root<Patient> patient = query.from(Patient.class);
+
+		// Only not deleted patient
+		Predicate deletedN = cb.equal(patient.get("deleted"), "N");
+		Predicate deletedNull = cb.isNull(patient.get("deleted"));
+		Predicate notDeleted = cb.or(deletedN, deletedNull);
+
+		List<Predicate> predicates = new ArrayList<Predicate>();
+		predicates.add(notDeleted);
+		for (Map.Entry<String, Object> entry : params.entrySet()) {
+			Path<String> keyPath = patient.get(entry.getKey());
+
+			if (entry.getKey().equals("birthDate")) {
+				Date birthDateFrom = (Date) entry.getValue();
+				Date birthDateTo = new Date(birthDateFrom.getTime() + (1000 * 60 * 60 * 24));
+				predicates.add(cb.between(keyPath.as(Date.class), birthDateFrom, birthDateTo));
+			} else {
+				if (entry.getValue() instanceof String) {
+					predicates.add(cb.like(keyPath, (String) entry.getValue() + "%"));
```

Comment:
I think we should implement case insensitive `like` as done in `wordExistsInOneOfPatientFields` method, eventually reusing the same code